homeassistant/components/google_domains.py

```diff
@@ -0,0 +1,77 @@
+"""
+Integrate with Google Domains.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/google_domains/
+"""
+import asyncio
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_DOMAIN, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'google_domains'
+
+INTERVAL = timedelta(minutes=5)
+
+UPDATE_URL = 'https://{}:{}@domains.google.com/nic/update'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DOMAIN): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Initialize the Google Domains component."""
+    domain = config[DOMAIN][CONF_DOMAIN]
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+
+    session = async_get_clientsession(hass)
+
+    result = yield from _update_google_domains(session, domain, user, password)
+
+    if not result:
+        return False
+
+    @asyncio.coroutine
+    def update_domain_interval(now):
+        """Update the Google Domains entry."""
+        yield from _update_google_domains(session, domain, user, password)
+
+    async_track_time_interval(hass, update_domain_interval, INTERVAL)
+
+    return result
+
+
+@asyncio.coroutine
+def _update_google_domains(session, domain, user, password):
+    """Update Google Domains."""
+    url = UPDATE_URL.format(user, password)
+
+    params = {
+        'hostname': domain
+    }
+
+    resp = yield from session.get(url, params=params)
```

Comment:
use `asyncio-timeout` like other components use i.e. microsoft_face

Reply:
Should I make this configurable like the MS Face one? And is 10s a good default here as well?

Reply:
This ended up being pretty simple so I just went ahead and make it configurable.